### DISCUSSION TITLE
Doc example for `setqflist` with `systemlist` doesn't work

### DISCUSSION THREAD

**Discussion by noahbald** [neovim/neovim#31020] [Q&A] [answered]
Hello, I'm looking through the docs and want to use something similar to this example for [`setqflist`](https://neovim.io/doc/user/quickfix.html#setqflist-examples)
```vim
:call setqflist([], ' ', {'lines' : systemlist('grep -Hn main *.c')})
```
```lua
-- or, in lua
vim.fn.setqflist({}, ' '. { lines = vim.fn.systemlist('grep -Hn main *.c') })
```

However when I run this nothing seems to happen. This is what it looks like when I run a similar grep from my shell.
```sh
# grep -Hn main ./**/*.nu
yazi/open_file.nu:3:def main [file_path: path] {
zellij/is_nvim_running.nu:3:export def main [list_clients_output: string] {
zellij/open_file.nu:5:def main [file_path: path, location?: string] {
zellij/open_file.nu:13:        | parse --regex '\w+\s+\w+\s+(?<rest>.*)'  # Use regex to match two words and capture the remaining text as 'rest'
zellij/quick_fix.nu:5:def main [tempfile: path] {
zellij/quick_fix.nu:13:        | parse --regex '\w+\s+\w+\s+(?<rest>.*)'  # Use regex to match two words and capture the remaining text as 'rest'
```

**Comment by noahbald** [comment, marked as answer]
Didn't realise I need to make an additional call to `:copen`. It was working fine, it's just that `setqflist` doesn't actually cause the quickfix list to appear 😅

**Comment by telemachus** [comment]
FYI, you can use the [`QuickFixCmdPost`](https://neovim.io/doc/user/autocmd.html#QuickFixCmdPost) event with an autocommand to call [`:cwindow`](https://neovim.io/doc/user/quickfix.html#%3Acwindow).
